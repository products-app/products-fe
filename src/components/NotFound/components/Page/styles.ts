export default {
  container: 'text-white flex flex-col items-center justify-center',
  iconNotFound: 'text-9xl',
  headingSpan: 'text-base300',
  textDescription: 'text-gray500',
  buttonAction: 'mt-3',
}
