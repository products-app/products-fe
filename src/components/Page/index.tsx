import Header from '../../components/Header'
import CartSidebar from '../CartSidebar'

const styles = {
  container: 'max-w-5xl mx-auto my-10',
}

type PageProps = {
  children: React.ReactNode
}

const Page = ({ children }: PageProps) => {
  return (
    <>
      <Header />
      <main>
        <div className={styles.container}>{children}</div>
      </main>
      <CartSidebar />
    </>
  )
}

export default Page
