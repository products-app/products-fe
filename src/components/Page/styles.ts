export default {
  container: 'max-w-5xl mx-auto my-10',
}
