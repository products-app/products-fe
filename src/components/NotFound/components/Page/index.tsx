import { LinkBreak } from 'phosphor-react'
import { Heading, Text, Button } from '@lebernardo/react'
import styles from './styles'
import { useNavigate } from 'react-router-dom'

const Page = () => {
  const navigate = useNavigate()

  const redirectToPageIndex = () => {
    navigate('/')
  }

  return (
    <div className={styles.container}>
      <LinkBreak className={styles.iconNotFound} />
      <Heading size="lg">404 - Página não encontrada</Heading>
      <Text size="lg" className={styles.textDescription}>
        Tente recarregar a página ou tentar mais tarde
      </Text>

      <Button
        variant="outline"
        className={styles.buttonAction}
        onClick={redirectToPageIndex}
      >
        Voltar para página inicial
      </Button>
    </div>
  )
}

export default Page
