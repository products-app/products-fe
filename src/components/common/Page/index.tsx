import Header from '@/components/common/Header'
import { CartSidebar } from '@/components/Cart'
import CheckoutModal from '@/components/Checkout/CheckoutModal'
import { ToastContainer } from 'react-toastify'
import { Container } from './styles'

type PageProps = {
  children?: React.ReactNode
}

const Page = ({ children }: PageProps) => {
  return (
    <>
      <Header />
      <main>
        <Container>{children}</Container>
      </main>
      <CartSidebar />
      <ToastContainer />
      <CheckoutModal />
    </>
  )
}

const defaultProps = {
  children: <></>,
}

Page.defaultProps = defaultProps
export default Page
