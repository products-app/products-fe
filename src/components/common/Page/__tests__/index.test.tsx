import { render, screen } from '@testing-library/react'
import Page from '../index'

describe('Loading component', () => {
  it('should render correctly', () => {
    const { container } = render(<Page>Test</Page>)

    expect(screen.getByText('Test')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
