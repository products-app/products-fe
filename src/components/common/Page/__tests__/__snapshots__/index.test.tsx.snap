// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render correctly 1`] = `
<div>
  <header
    class="c-clOfPL"
  >
    <div
      class="c-iwTvpr"
    >
      <a
        href="/"
      >
        Prompt
      </a>
      <div
        class="c-eGMMdN"
      >
        <div
          class="c-bKZIAC"
        >
          <input
            class="c-bgjSgL"
            id="search"
            placeholder="Pesquise por produtos"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="c-cGenGQ"
      >
        <button
          aria-label="button-redirect-login"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <circle
              cx="128"
              cy="128"
              fill="none"
              r="96"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
            <circle
              cx="128"
              cy="120"
              fill="none"
              r="40"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
            <path
              d="M63.8,199.4a72,72,0,0,1,128.4,0"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
        </button>
        <button
          aria-label="button-toggle-sidebar"
          data-cy="button-cart"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <circle
              cx="80"
              cy="216"
              r="16"
            />
            <circle
              cx="184"
              cy="216"
              r="16"
            />
            <path
              d="M42.3,72H221.7l-26.4,92.4A15.9,15.9,0,0,1,179.9,176H84.1a15.9,15.9,0,0,1-15.4-11.6L32.5,37.8A8,8,0,0,0,24.8,32H8"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
          <div
            class="c-egqsFx"
          >
            0
          </div>
        </button>
      </div>
    </div>
  </header>
  <main>
    <div
      class="c-cyakpB"
    >
      Test
    </div>
  </main>
  <section
    class="c-hEQxdV"
    id="sidebar"
    style="transform: translateX(100%);"
  >
    <button
      aria-label="button-close-sidebar"
      class="c-eOLtzh"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <line
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="200"
          x2="56"
          y1="56"
          y2="200"
        />
        <line
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="200"
          x2="56"
          y1="200"
          y2="56"
        />
      </svg>
    </button>
    <nav
      class="c-nMCuV"
    >
      <div
        class="c-gzevCw"
      >
        <p
          class="c-dgMvMW c-fMsmQc c-dgMvMW-eQRsLC-size-lg"
        >
          Carrinho de compras
        </p>
        <div
          class="c-jwMKId"
          id="cart-sidebar-items"
        >
          <div
            class="c-dzUtkR"
          >
            <img
              alt="Imagem de carrinho vazio"
              data-nimg="1"
              decoding="async"
              height="0"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fempty-cart.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fempty-cart.png&w=640&q=75 640w, /_next/image?url=%2Fempty-cart.png&w=750&q=75 750w, /_next/image?url=%2Fempty-cart.png&w=828&q=75 828w, /_next/image?url=%2Fempty-cart.png&w=1080&q=75 1080w, /_next/image?url=%2Fempty-cart.png&w=1200&q=75 1200w, /_next/image?url=%2Fempty-cart.png&w=1920&q=75 1920w, /_next/image?url=%2Fempty-cart.png&w=2048&q=75 2048w, /_next/image?url=%2Fempty-cart.png&w=3840&q=75 3840w"
              style="color: transparent;"
              width="0"
            />
            <p
              class="c-dgMvMW c-igsGei c-dgMvMW-kWyXas-size-md"
            >
              Adicione algo em seu carrinho...
            </p>
          </div>
        </div>
      </div>
    </nav>
  </section>
  <div
    class="Toastify"
  />
  <div />
</div>
`;
